aws/data_source_aws_db_subnet_group.go

```diff
@@ -0,0 +1,93 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsDbSubnetGroup() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDbSubnetGroupRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"subnets": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsDbSubnetGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	opts := &rds.DescribeDBSubnetGroupsInput{
+		DBSubnetGroupName: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Reading DB SubnetGroup: %s", opts)
+
+	resp, err := conn.DescribeDBSubnetGroups(opts)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.DBSubnetGroups) < 1 {
```

Comment:
I believe these 2 length checks can be reduced to a check like in the resource https://github.com/terraform-providers/terraform-provider-aws/blob/b009f7ac157bdea33c902f8070d1cbcb13cf428b/aws/resource_aws_db_subnet_group.go#L122
as the search is by `name` and the aws SDK only allows 1 subnet group to use a particular name (error 400 thrown at Create time https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBSubnetGroup.html)